addon/components/au-time-picker.js

```diff
@@ -28,103 +47,44 @@ export default class AuTimePickerComponent extends Component {
     return isPresent(this.args.showNow) ? this.args.showNow : true;
   }
 
-  /*
-   * Increments or decrements a time value.
-   * HourValue has a max of 23 while minute & seconds have a max of 59
-   * HourValue, minute & seconds have a min of 0
-   */
-
   @action
   increment(elem) {
-    if (elem == 'hourValue') {
-      ++this[elem];
-
-      if (this[elem] >= 23) {
-        this[elem] = 23;
-      }
-    }
-
-    if (elem != 'hourValue') {
-      ++this[elem];
-
-      if (this[elem] >= 59) {
-        this[elem] = 59;
-      }
-    }
+    this[elem] = this.validateTimeValue(this[elem] + 1, elem);
```

Comment:
While testing out the component in Storybook I noticed that something funky was going on when incrementing a value when the original value was passed in. If for example I set "hours" to `11` in Storybook and increment it, the value jumps to 23. This is because Storybook passes it in as a string, so `this[elem] + 1` will result in `111`, which will set it to the max value after that by the validateTimeValue method.

I think this partly a Storybook configuration issue. It should pass the values as integers, but maybe the arguments need to be validated as well in case consumers do something similar? Maybe an assert should be thrown if the initial argument values are strings? That will probably catch most "misuses".

Reply:
It seems the Storybook side can be resolved by changing the control type to `number` for the `hours`, `minutes` and `seconds` args and setting the default value to a number instead of a string. That seemed to work here at least. The Storybook docs aren't very clear about this.

Reply:
Right, I saw that too, but then forgot about that! This needs to be addressed.

Reply:
This can be solved with `trackedReset`. See other comment about this. Basically, we could validate every parameter value, even on changes.